src/jquery.multiselect.js

```diff
@@ -90,110 +140,130 @@
+      if (!options.listbox) {
+         // The button that opens the widget menu.  Note that this is inserted later below.
+         var $button = ( this.$button = $( document.createElement('button') ) )
+               .addClass('ui-multiselect ui-widget ui-state-default ui-corner-all'
+                           + (/\bbutton\b/i.test(wrapText) ? '' : ' ui-multiselect-nowrap')
+                           + (classes ? ' ' + classes : '')
+                           )
+               .attr({
+                 'type': 'button',
+                 'title': elSelect.title,
+                 'tabIndex': elSelect.tabIndex,
+                 'id': elSelect.id ? elSelect.id  + '_ms' : null
+               })
+               .prop('aria-haspopup', true)
+               .html( _linkHTML('<span class="{{class}}" title="{{title}}">{{icon}}</span>', 'open') );
+
+         this.$buttonlabel = $( document.createElement('span') )
+               .html(options.noneSelectedText || $element[0].placeholder)
+               .appendTo( $button );
+      }
+
+      // Header controls will contain the links & ordering specified by the header option.
       // Depending on how the options are set, this may be empty or simply plain text
-      var headerLinksHTML = ( headerOn === true
-               ?  (checkAllText === null ? '' : '<li><a class="ui-multiselect-all" title="Check All">' + iconSet.checkAll + (checkAllText ? '<span>' + checkAllText + '</span>' : '' ) + '</a></li>')
-                  + (uncheckAllText === null ? '' : '<li><a class="ui-multiselect-none" title="Uncheck All">' + iconSet.uncheckAll + (uncheckAllText ? '<span>' + uncheckAllText + '</span>' : '' ) + '</a></li>')
-                  + (flipAllText === null ? '' : '<li><a class="ui-multiselect-flip" title="Flip All">' + iconSet.flipAll + '<span>' + (flipAllText ? '<span>' + flipAllText + '</span>' : '' ) + '</a></li>')
-               :  (typeof headerOn === 'string' ? '<li>' + headerOn + '</li>' : '') );
+      var headerLinksHTML = '';
+      if (options.header === false) {}  // no-op for no header
+      else if (options.header.charAt(0) === '=') {
+         headerLinksHTML = '<li>' + options.header.slice(1) + '</li>';
+      }
+      else {
+         var hdrLinkTemplate = '<li><a class="{{class}}" title="{{title}}">{{icon}}<span>{{text}}</span></a></li>',
+               headerLinks = options.header
+                              .replace(/all\b/g,'All')
+                              .replace(options.maxSelected ? /\bcheckAll\b/ : '', '')
+                              .replace(/\s|^,+|,+$/g,'')
+                              .split(',');
+         for (var x = 0, len = headerLinks.length; x < len; x++) {
+            if (headerLinks[x] && headerLinks[x] in this.linkInfo
+                && !( headerLinks[x] in {'open':1, 'close':1, 'collapse':1, 'expand':1} ) ) {
+               headerLinksHTML += _linkHTML(hdrLinkTemplate, headerLinks[x]).replace(/<span><\/span/ig, '');
```

Comment:
I'm not okay with the fact that the generated html has to then be fixed with an additional regex replace.

Reply:
That is just for the case of no text for the link in the linkInfo object.  I could maybe just do a ternary to switch to a different template for the no text case.

Reply:
`_linkHTML` should probably handle that logic instead.

Reply:
@mlh758 Ok, I had it there originally, then moved it out.  lol

Reply:
Here's another thought. You mentioned the mess of string concatenation...

Would using jQuery to build the template up make it easier/cleaner? Something like is done in makeOption or buildOptionList?

It's another more code/maybe less complicated trade off. As long as `_linkHTML` handles it and produces valid HTML though I'll be happy.

Reply:
@mlh758 Ok, let me chew on this a bit.   As I am sure you are aware now, I do like speedy code, but sometimes it is worthwhile to strategically sacrifice some speed for cleaner, easier to maintain code.

The string concatenation would likely be fastest.  However, once you get to a certain number of header links, it gets really messy.   I made the decision to try for some cleaner code because building the header up is only done in the _create() "once", not in a loop or anything where you would notice much of a speed difference.  The only place this may matter is if you are doing a lot of widget instances (like I do in my web app), but I think I could live with a small slow-down if it is minimal and makes for cleaner code.

What I would like to see is the following:
1.  Allow the order of header links to be specified and specify which to include.
2.  Have a reasonable way to translate the info in the linkInfo/linkDefaults object into header links without a lot of messy code to do it.  I went with a template approach, but open to alternatives.
3.  Allow for the header to be shut off or custom header text to be specified, but recognize that these are likely edge cases, so maybe they do not have to be as convenient.
4.  Keep the # of options to set up the header to a minimum, if at all possible.

If we can arrive at a way to accomplish the above, I will be fine with that.

Reply:
@mlh758 One possibility for the header option is to use an array instead of a comma separated string for specifying the displayed links and link order.

Thus, if header === false, no header

If typeof header === array, then process as which links to show and what order,  The array elements are the keys for the linkInfo object to indicate which links and which order.  No string clean up/oops handling needed either.

If typeof header === string, then show custom text.  (No = prefix needed for this case for this approach.)

Reply:
Hey I like that last one a lot. Then we don't need people to learn the '=' syntax, it's just using a string for an absolute value like before. The array for the list let's us skip the split and trim step completely, and anything falsey just shows no header. That's perfect.

Reply:
@mlh758 Ok, as the array elements would be the keys for the linkInfo object, they would be case-sensitive.  e.g. `['checkall' ,'uncheckall']` will not work.

Reply:
Sounds great. I'd prefer typos cause errors as early as possible anyway.

Reply:
@mlh758 I have switched to the falsey/string/array header, and I am using your filter/map approach, also.

However, you still will have to call .html() because of the icons are specified as html elements.  I could build the links up w/ jQuery to get rid of the _linkHTML() function, but I will still be calling html anyway.  Because of this, I sort of with with the template approach.  However, I am opened to being convinced otherwise.  ;-)